homeassistant/components/fan/insteon_local.py

```diff
@@ -31,85 +28,30 @@
     """Set up the Insteon local fan platform."""
     insteonhub = hass.data['insteon_local']
-
-    conf_fans = load_json(hass.config.path(INSTEON_LOCAL_FANS_CONF))
-    if conf_fans:
-        for device_id in conf_fans:
-            setup_fan(device_id, conf_fans[device_id], insteonhub, hass,
-                      add_devices)
-
-    else:
-        linked = insteonhub.get_linked()
-
-        for device_id in linked:
-            if (linked[device_id]['cat_type'] == 'dimmer' and
-                    linked[device_id]['sku'] == '2475F' and
-                    device_id not in conf_fans):
-                request_configuration(device_id,
-                                      insteonhub,
-                                      linked[device_id]['model_name'] + ' ' +
-                                      linked[device_id]['sku'],
-                                      hass, add_devices)
-
-
-def request_configuration(device_id, insteonhub, model, hass,
-                          add_devices_callback):
-    """Request configuration steps from the user."""
-    configurator = hass.components.configurator
-
-    # We got an error if this method is called while we are configuring
-    if device_id in _CONFIGURING:
-        configurator.notify_errors(
-            _CONFIGURING[device_id], 'Failed to register, please try again.')
-
+    if discovery_info is None:
         return
 
-    def insteon_fan_config_callback(data):
-        """The actions to do when our configuration callback is called."""
-        setup_fan(device_id, data.get('name'), insteonhub, hass,
-                  add_devices_callback)
-
-    _CONFIGURING[device_id] = configurator.request_config(
-        'Insteon  ' + model + ' addr: ' + device_id,
-        insteon_fan_config_callback,
-        description=('Enter a name for ' + model + ' Fan addr: ' + device_id),
-        entity_picture='/static/images/config_insteon.png',
-        submit_caption='Confirm',
-        fields=[{'id': 'name', 'name': 'Name', 'type': ''}]
-    )
-
-
-def setup_fan(device_id, name, insteonhub, hass, add_devices_callback):
-    """Set up the fan."""
-    if device_id in _CONFIGURING:
-        request_id = _CONFIGURING.pop(device_id)
-        configurator = hass.components.configurator
-        configurator.request_done(request_id)
-        _LOGGER.info("Device configuration done!")
-
-    conf_fans = load_json(hass.config.path(INSTEON_LOCAL_FANS_CONF))
-    if device_id not in conf_fans:
-        conf_fans[device_id] = name
-
-    save_json(hass.config.path(INSTEON_LOCAL_FANS_CONF), conf_fans)
+    linked = discovery_info['linked']
 
-    device = insteonhub.fan(device_id)
-    add_devices_callback([InsteonLocalFanDevice(device, name)])
+    for device_id in linked:
+        if (linked[device_id]['cat_type'] == 'dimmer' and
+                linked[device_id]['sku'] == '2475F'):
+            device = insteonhub.fan(device_id)
+            add_devices([InsteonLocalFanDevice(device)])
```

Comment:
Please collect all devices in a list before adding them, so you only make one call to `add_devices`. Sorry about not catching that the first time.